src/app/store/accounts/blockchain/bitcoin/taproot-account.hooks.ts

```diff
@@ -23,13 +31,78 @@ function useTaprootKeychainByAccount() {
 
 export function useCurrentTaprootAccountKeychain() {
   const currentAccountIndex = useCurrentAccountIndex();
-  const accountKeychain = useTaprootKeychainByAccount();
-  if (!accountKeychain) throw new Error('No account keychain found');
-  return accountKeychain(currentAccountIndex);
+  const accountKeychain = useTaprootCurrentNetworkAccountKeychain();
+  if (!accountKeychain) return; // TODO: Revisit this return early
+  const keychain = accountKeychain(currentAccountIndex);
+  if (!keychain) throw new Error('No account keychain found');
+  return keychain;
+}
+
+// Concept of current address index won't exist with privacy mode
+export function useCurrentTaprootAddressIndexKeychain() {
+  const keychain = useCurrentTaprootAccountKeychain();
+  if (!keychain) return; // TODO: Revisit this return early
+  return deriveAddressIndexZeroFromAccount(keychain);
+}
+
+export function useSignBitcoinTaprootTx() {
+  const index = useCurrentAccountIndex();
+  const keychain = useTaprootCurrentNetworkAccountKeychain()?.(index);
+
+  return useCallback(
+    (tx: btc.Transaction) => {
+      if (isUndefined(keychain)) return;
+      tx.sign(deriveAddressIndexZeroFromAccount(keychain).privateKey!);
+    },
+    [keychain]
+  );
+}
+
+interface UseSignBitcoinTaprootInputAtIndexArgs {
+  allowedSighash?: btc.SignatureHash[];
+  idx: number;
+  tx: btc.Transaction;
+}
+export function useSignBitcoinTaprootInputAtIndex() {
+  const index = useCurrentAccountIndex();
+  const keychain = useTaprootCurrentNetworkAccountKeychain()?.(index);
+
+  return useCallback(
+    ({ allowedSighash, idx, tx }: UseSignBitcoinTaprootInputAtIndexArgs) => {
+      if (isUndefined(keychain)) return;
+      tx.signIdx(deriveAddressIndexZeroFromAccount(keychain).privateKey!, idx, allowedSighash);
+    },
+    [keychain]
+  );
+}
+
+export function useAllBitcoinTaprootNetworksByAccount() {
+  const mainnetKeychainAtAccount = useSelector(selectMainnetTaprootKeychain);
+  const testnetKeychainAtAccount = useSelector(selectTestnetTaprootKeychain);
+
+  return useCallback(
+    (accountIndex: number) => {
+      if (!mainnetKeychainAtAccount || !testnetKeychainAtAccount)
+        throw new Error('Cannot derive addresses in non-software mode');
+      return {
+        mainnet: deriveTaprootReceiveAddressIndex({
+          xpub: mainnetKeychainAtAccount(accountIndex).publicExtendedKey,
+          index: accountIndex,
+          network: 'mainnet',
+        })?.address,
```

Comment:
⚠️ passing account index as address index

Reply:
I believe we can just pass `0` here for now as we are just providing the first address?

Reply:
Tested and works with hardcoding `0`, left a comment in the code that it is only temporary until the address api is ready. If there is a better temp solution, let me know or fix before me tom.